### PR TITLE
Allow volumes and volume mounts to be defined as Dicts

### PR DESCRIPTION
**Pull request by sunu**
I've made the following changes:

- Modified the trait type for volumes and volume mounts to `Union(trait_types=[List(), Dict()])` so that they can be defined as dictionaries (for easy overriding) or as lists (for backwards compatibility)
- Updated the help messages explaining both list and dictionary input options and clarifying how dictionary inputs are processed (sorted lexicographically by keys)
- Added tests for defining volumes and volume mounts as dictionaries and test that the output is sorted lexicographically and tests for defining volumes and volume mounts as lists to make sure these changes are backwards compatible

Defining these values as dictionaries enables easy overriding. refs https://github.com/jupyterhub/jupyterhub/pull/4822 and https://github.com/NASA-IMPACT/veda-jupyterhub/issues/44

### CODE REVIEW

**Comment by sunu** [comment]
@yuvipanda Could you please take a look at these changes and let me know if they look good? If this looks ok, I'll proceed to make similar updates for other traitlets that need to support dictionaries. Thanks!

File: kubespawner/spawner.py [hunk 1]
```diff
@@ -1869,6 +1891,12 @@ def _expand_all(self, src):
         else:
             return src
 
+    def _sorted_dict_values(self, src):
```
Comment (yuvipanda) [inline kubespawner/spawner.py:1894]: Can you add a docstring here?

**Review by yuvipanda — APPROVED** [review]
One minor ask for a docstring, but otherwise this approach looks great to me.

Thank you for working on this, @sunu!

File: kubespawner/spawner.py (hunk 1 above)
Comment (sunu) [reply kubespawner/spawner.py:1894]: Done!

**Comment by sunu** [comment]
Thanks for the review @yuvipanda! I have added the missing docstring. I'll open a follow up PR to make similar updates for other traitlets that need to support dictionaries.

**Comment by yuvipanda** [comment]
Thanks @sunu!

**Pull request merged**